### DISCUSSION TITLE
AutoCompleteBox: styling dropdown/popup content

### DISCUSSION THREAD

**Discussion by ManuelKetisch** [AvaloniaUI/Avalonia#13364] [Q&A] [answered]
Hello,

I want to change the foreground color for the DropDown/Popup part of the AutoCompleteBox control.
Currently the text presented in the DropDown/Popup when typing is a very light gray, that is hardly readable on a white background.

I tried a few things, but it did not work.
`
<Style Selector="AutoCompleteBox /template/ ContentPresenter#PART_ContentPresenter">
    <Setter Property="Foreground" Value="#2d2a2e" />
</Style>

<Style Selector="AutoCompleteBox /template/ TextBlock#PART_Popup">
    <Setter Property="WindowManagerAddShadowHint" Value="False"/>
</Style>
`

From what i found on this page, it looks to me, that the foreground of the popup can not be styled.
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/AutoCompleteBox.xaml

Btw. i am using a DropDown Behavior from this discussion, i case this would mess with the styling of the control.
[https://github.com/AvaloniaUI/Avalonia/discussions/13301](url)

I am fairly new to Avalonia so maybe i am just missing something.
Any help would be appreciated.

**Comment by timunie** [comment, marked as answer]
TextBlock#PART_Popup is not a valid selector. Please read selector docs. But normally I'd think something leaks into your style as it should not be light gray normally. Try to make a minimal sample on your own and try to figure out what went wrong by copy over the relevant parts

https://docs.avaloniaui.net/docs/next/reference/styles/style-selector-syntax

**Reply by ManuelKetisch** [reply]
Ok, i found the problem. 
Your suggestion that another style was leaking into the style of the AutoCompleteBox was correct.
The culprit is the styling for my TextBlocks. I am using a light gray color for those, because i have a dark gray background for my app.
It seems the AutoCompleteBox uses TextBlocks for the popup suggestions.

So to fix this, i should probably use style classes, to prevent the AutoCompleteBoxes from using this particular style for themself.
I will try to figure out how to do that. 

Thanks for helping out!

**Reply by ManuelKetisch** [reply]
Found a way to fix it using TextBlock as "Descendant Operator"

`
        
        <Style Selector="AutoCompleteBox TextBlock">
            <Setter Property="Foreground" Value="#2d2a2e" />
        </Style>
`

Thanks again for the assistance.
